sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java

```diff
@@ -0,0 +1,153 @@
+            try {
+                T t = type.newInstance();
+
+                Map<String, Field> declaredFieldMapping = Arrays.stream(type.getDeclaredFields())
+                        .collect(Collectors.toMap(field -> field.getName().toLowerCase(Locale.ROOT), field -> field));
+
+                tableRow.getRow().stream()
+                        .forEach(tableCell -> {
+                            String columnName = tableCell.getColumnName();
+                            try {
+                                Field field = declaredFieldMapping.get(columnName.toLowerCase(Locale.ROOT));
+                                if (field == null) {
+                                    return;
+                                }
+                                field.setAccessible(true);
+                                if (tableCell.getColumnType() == LogsColumnType.BOOL) {
+                                    field.set(t, tableCell.getValueAsBoolean());
+                                } else if (tableCell.getColumnType() == LogsColumnType.DATETIME) {
+                                    field.set(t, tableCell.getValueAsDateTime());
+                                } else if (tableCell.getColumnType() == LogsColumnType.DYNAMIC) {
+                                    if (tableCell.getValueAsDynamic() != null) {
+                                        field.set(t,
+                                                tableCell.getValueAsDynamic()
+                                                        .toObject(TypeReference.createInstance(field.getType())));
+                                    }
+                                } else if (tableCell.getColumnType() == LogsColumnType.INT) {
+                                    field.set(t, tableCell.getValueAsInteger());
+                                } else if (tableCell.getColumnType() == LogsColumnType.LONG) {
+                                    field.set(t, tableCell.getValueAsLong());
+                                } else if (tableCell.getColumnType() == LogsColumnType.REAL
+                                        || tableCell.getColumnType() == LogsColumnType.DECIMAL) {
+                                    field.set(t, tableCell.getValueAsDouble());
+                                } else if (tableCell.getColumnType() == LogsColumnType.STRING
+                                        || tableCell.getColumnType() == LogsColumnType.GUID
+                                        || tableCell.getColumnType() == LogsColumnType.TIMESPAN) {
+                                    field.set(t, tableCell.getValueAsString());
+                                }
+                                field.setAccessible(false);
+                            } catch (IllegalAccessException ex) {
+                                throw LOGGER.logExceptionAsError(
+                                        new IllegalArgumentException("Failed to set column value for " + columnName, ex));
```

Comment:
Why allow partial creation of the object?

Reply:
This won't create partial objects. If we fail to set the value of any column, this whole method will throw. So, the caller of this method will not get any result.